elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/PredicateAnalyzer.java

```diff
@@ -117,7 +117,7 @@ static QueryBuilder analyze(RexNode expression) throws ExpressionNotAnalyzableEx
       }
       return e != null ? e.builder() : null;
     } catch (Throwable e) {
-      Throwables.propagateIfPossible(e, UnsupportedOperationException.class);
+      Throwables.throwIfInstanceOf(e, UnsupportedOperationException.class);
```

Comment:
According to the Javadoc, throwIfInstanceOf is not a 1:1 replacement for propagateIfPossible.

The old code would throw other RunTime exceptions directly, while the new one wraps them.

This may be a good thing, but the commit message wording does not suggest behavioural changes.

Reply:
I wasn't aware that the methods have different behavior. Thanks for bringing it to my attention.

However, there is no obligation to document behavioral changes that occur in 'behavior is unspecified' territory.

This block - like pretty much every `catch (Throwable)` block - is in that territory, and I didn't think too much about which use cases would be affected because - by definition - there are no documented, tested use cases.